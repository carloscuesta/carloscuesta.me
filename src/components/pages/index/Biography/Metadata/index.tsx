import { useState, useEffect } from 'react'

const Metadata = () => {
  const [date, setDate] = useState<string>()
  new Date().toLocaleTimeString('es-ES', { timeZone: 'Europe/Madrid' })

  useEffect(() => {
    const timer = setInterval(() => {
      return setDate(
        new Date().toLocaleTimeString('en-DE', { timeZone: 'Europe/Madrid' })
      )
    }, 500)

    return () => clearInterval(timer)
  }, [])

  return (
    <div className="flex text-sm lowercase opacity-60">
      <span className="block w-[62px] whitespace-nowrap">{date}</span>
      <div className="flex gap-1.5">
        <span>Â·</span>
        <a
          target="_blank"
          rel="noopener noreferrer nofollow"
          href="https://maps.apple.com/place?q=Barcelona&ll=41.3826807%2C2.1770239&auid=2262861905204318884&lsp=7618&address=Barcelona%2C%20Spain"
        >
          Barcelona, Spain
        </a>
      </div>
    </div>
  )
}

export default Metadata
