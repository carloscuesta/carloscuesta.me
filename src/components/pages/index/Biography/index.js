// @flow
import React from 'react'

import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Biography = () => (
  <section className={styles.biography}>
    <Wrapper>
      <div className='row middle-xs'>
        <div className='col-xs-12 col-sm-6'>
          <h3 className={styles.biographyText}>
            I'm a 24 years old <b>Front End Engineer</b> based in <b>Barcelona</b>,
            working at <a href='https://ulabox.com' rel='noopener noreferrer'><b>Ulabox</b></a>.
            I'm a coding passionate that is focused on <b>JavaScript</b>, specifically in React & React-Native.
            Interested in design, tech and <b>OpenSource</b> ❤️
          </h3>

          <h3 className={styles.biographyText}>
            You can find me on{' '}
            <a href='https://github.com/carloscuesta' target='_blank' rel='noopener noreferrer'>GitHub</a>,{' '}
            <a href='https://twitter.com/crloscuesta' target='_blank' rel='noopener noreferrer'>Twitter</a>{' and '}
            <a href='https://linkedin.com/in/crloscuesta' target='_blank' rel='noopener noreferrer'>LinkedIn</a>
          </h3>
        </div>

        <div className='col-xs-12 col-sm-6'>
          <img
            alt='Carlos Cuesta'
            className={styles.profilePicture}
            src='images/carloscuesta.jpg'
          />
        </div>
      </div>
    </Wrapper>

  </section>
)

export default Biography
