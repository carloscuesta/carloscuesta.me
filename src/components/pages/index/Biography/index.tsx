import Wrapper from 'src/components/shared/Wrapper'
import styles from './styles.module.css'

const Biography = () => (
  <section>
    <Wrapper>
      <div className={styles.container}>
        <div className={styles.biographyContainer}>
          <p className={styles.text}>
            I&#39;m a Front End Engineer interested in{' '}
            <b>design</b>, <b>under engineering</b> and <b>open source</b>.
            I work at <a target='_blank' rel='noopener noreferrer' href='https://n26.com'>N26</a>.
          </p>

          <p className={styles.text}>
            I&#39;m focused on building products with JavaScript, specifically React and React-Native.
          </p>

          <p className={styles.text}>
            When I&#39;m not coding, I&#39;m likely at the gym or surfing the city with my longboard.
          </p>

          <p className={styles.text}>
            You can find me on{' '}
            <a href='https://github.com/carloscuesta' target='_blank' rel='noopener noreferrer'>GitHub</a>,{' '}
            <a href='https://twitter.com/crloscuesta' target='_blank' rel='noopener noreferrer'>Twitter</a>{' and '}
            <a href='https://linkedin.com/in/crloscuesta' target='_blank' rel='noopener noreferrer'>LinkedIn</a>.
          </p>
        </div>

        <div className={styles.pictureContainer}>
          <img
            alt='Carlos Cuesta'
            className={`lazyload ${styles.picture}`}
            data-src='https://res.cloudinary.com/carloscuesta/image/upload/s--ekQN9GP_--/c_scale,w_320/v1594588508/carloscuesta.jpg'
            src='https://res.cloudinary.com/carloscuesta/image/upload/s--rq7ZUVpY--/c_scale,e_blur:100,h_120,q_10,w_120/v1594588508/carloscuesta.jpg'
          />
        </div>

      </div>
    </Wrapper>
  </section>
)

export default Biography
