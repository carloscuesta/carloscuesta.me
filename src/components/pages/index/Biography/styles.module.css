.text {
  color: var(--text);
  font-size: 1.25rem;
  font-weight: 400;
}

.picture {
  border-radius: 50%;
  display: flex;
  filter: blur(8px);
  height: auto;
  margin-left: auto;
  max-width: 100%;
  transition: filter 200ms ease-in-out;
  width: 360px;
}

.picture:global(.lazyloaded) {
  filter: blur(0);
}

@media(max-width: 768px) {
  .picture {
    margin: 0 auto;
  }
}
