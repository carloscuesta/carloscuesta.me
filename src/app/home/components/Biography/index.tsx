import NextLink from 'next/link'
import Image from 'next/image'

import Wrapper from 'src/components/shared/Wrapper'
import Metadata from './Metadata'

const Link = (props: { href: string; label: string }) => {
  const linkClassName = 'underline-offset-2 underline'

  if (props.href.startsWith('/')) {
    ;<NextLink href={props.href} className={linkClassName}>
      {props.label}
    </NextLink>
  }

  return (
    <a
      target="_blank"
      href={props.href}
      className={linkClassName}
      rel="noopener noreferrer"
    >
      {props.label}
    </a>
  )
}

const Biography = () => (
  <section className="my-12">
    <Wrapper>
      <Metadata />

      <div className="flex justify-between gap-8">
        <div>
          <h1 className="flex gap-2 pb-4 text-2xl">
            <span className="font-bold">Hey, I'm Carlos Cuesta</span>
            <span className="inline-block origin-[70%_70%] animate-wave">
              👋🏼
            </span>
          </h1>

          <div className="grid gap-4">
            <p className="sm:max-w-md">
              I'm a Front End Engineer based in Barcelona. I{' '}
              <Link href="https://github.com/carloscuesta" label="code" />,{' '}
              <Link href="/blog" label="write" /> and build stuff on internet.
            </p>

            <p className="sm:max-w-md">
              I love working in-between product, engineering and developer
              experience, currently at{' '}
              <Link href="https://n26.com" label="N26" />.
            </p>
          </div>
        </div>

        <Image
          alt="Carlos Cuesta"
          src="https://res.cloudinary.com/carloscuesta/image/upload/s--ekQN9GP_--/c_scale,w_320/v1594588508/carloscuesta.jpg"
          placeholder="blur"
          blurDataURL="data:image/webp;base64,UklGRvQCAABXRUJQVlA4WAoAAAAgAAAAOQAAOQAASUNDUMgBAAAAAAHIAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAACRyWFlaAAABFAAAABRnWFlaAAABKAAAABRiWFlaAAABPAAAABR3dHB0AAABUAAAABRyVFJDAAABZAAAAChnVFJDAAABZAAAAChiVFJDAAABZAAAAChjcHJ0AAABjAAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAAgAAAAcAHMAUgBHAEJYWVogAAAAAAAAb6IAADj1AAADkFhZWiAAAAAAAABimQAAt4UAABjaWFlaIAAAAAAAACSgAAAPhAAAts9YWVogAAAAAAAA9tYAAQAAAADTLXBhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABtbHVjAAAAAAAAAAEAAAAMZW5VUwAAACAAAAAcAEcAbwBvAGcAbABlACAASQBuAGMALgAgADIAMAAxADZWUDggBgEAAJALAJ0BKjoAOgA+bTCSRiQjIaEwFgkAgA2JQBWHOBs/zMoLISaWACRRJkfyBI4Lxdd/xd03yJpIGFTUqOQGP4xGt8WUYj9jng+v+2wjcftkQSQUlbocxVkWWLjln0a0IfovTZIAAP70062J9GjrH7g8QW5d3F+LQ8rXmHN3okrCe1/lyHE1N4J7O4QOlbIkk3Z32UNX6AwqF01B+DCejpwmvHfxAntOE/kxlbpt/hamu8MLs/k+nwpMp4g1BIvX+zQnii+y1Kl3Frd5sNA18WL4mwthjxSBIjPSiiMNEjmgFriUlUlYCmgWI0KnfUFVrqpG5rSDeBN77cUw84dmPu7Q3pMAAAA="
          className="hidden h-44 w-44 transform-gpu rounded-full sm:block"
          width={176}
          height={176}
        />
      </div>
    </Wrapper>
  </section>
)

export default Biography
