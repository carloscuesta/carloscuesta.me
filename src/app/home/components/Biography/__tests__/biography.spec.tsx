import { render } from '@testing-library/react'

import Biography from '../index'

describe('Biography', () => {
  it('should match Biography component', () => {
    const { container } = render(<Biography />)

    expect(container).toMatchSnapshot()
  })
})
