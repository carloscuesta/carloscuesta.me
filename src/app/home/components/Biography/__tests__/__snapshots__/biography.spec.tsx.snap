// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Biography should match Biography component 1`] = `
<section
  className="my-12"
>
  <div
    className="m-[0_auto] max-w-4xl px-6 sm:px-4 "
  >
    <div
      className="flex text-sm lowercase opacity-60"
    >
      <span
        className="block w-[62px] whitespace-nowrap"
      />
      <div
        className="flex gap-1.5"
      >
        <span>
          ¬∑
        </span>
        <a
          href="https://maps.apple.com/place?q=Barcelona&ll=41.3826807%2C2.1770239&auid=2262861905204318884&lsp=7618&address=Barcelona%2C%20Spain"
          rel="noopener noreferrer nofollow"
          target="_blank"
        >
          Barcelona, Spain
        </a>
      </div>
    </div>
    <div
      className="flex justify-between gap-8"
    >
      <div>
        <h1
          className="flex gap-2 pb-4 text-2xl"
        >
          <span
            className="font-bold"
          >
            Hey, I'm Carlos Cuesta
          </span>
          <span
            className="inline-block origin-[70%_70%] animate-wave"
          >
            üëãüèº
          </span>
        </h1>
        <div
          className="grid gap-4"
        >
          <p
            className="sm:max-w-md"
          >
            I'm a Front End Engineer based in Barcelona. I
             
            <a
              className="underline-offset-2 underline"
              href="https://github.com/carloscuesta"
              rel="noopener noreferrer"
              target="_blank"
            >
              code
            </a>
            ,
             
            <a
              className="underline-offset-2 underline"
              href="/blog"
              rel="noopener noreferrer"
              target="_blank"
            >
              write
            </a>
             and build stuff on internet.
          </p>
          <p
            className="sm:max-w-md"
          >
            I love working in-between product, engineering and developer experience, currently at
             
            <a
              className="underline-offset-2 underline"
              href="https://n26.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              N26
            </a>
            .
          </p>
        </div>
      </div>
      <img
        alt="Carlos Cuesta"
        className="hidden h-44 w-44 transform-gpu rounded-full sm:block"
        data-nimg="1"
        decoding="async"
        height={176}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fs--ekQN9GP_--%2Fc_scale%2Cw_320%2Fv1594588508%2Fcarloscuesta.jpg&w=384&q=75"
        srcSet="/_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fs--ekQN9GP_--%2Fc_scale%2Cw_320%2Fv1594588508%2Fcarloscuesta.jpg&w=256&q=75 1x, /_next/image?url=https%3A%2F%2Fres.cloudinary.com%2Fcarloscuesta%2Fimage%2Fupload%2Fs--ekQN9GP_--%2Fc_scale%2Cw_320%2Fv1594588508%2Fcarloscuesta.jpg&w=384&q=75 2x"
        style={
          {
            "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 176 176'%3E%3Cfilter id='b' color-interpolation-filters='sRGB'%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3CfeColorMatrix values='1 0 0 0 0 0 1 0 0 0 0 0 1 0 0 0 0 0 100 -1' result='s'/%3E%3CfeFlood x='0' y='0' width='100%25' height='100%25'/%3E%3CfeComposite operator='out' in='s'/%3E%3CfeComposite in2='SourceGraphic'/%3E%3CfeGaussianBlur stdDeviation='20'/%3E%3C/filter%3E%3Cimage width='100%25' height='100%25' x='0' y='0' preserveAspectRatio='none' style='filter: url(%23b);' href='data:image/webp;base64,UklGRvQCAABXRUJQVlA4WAoAAAAgAAAAOQAAOQAASUNDUMgBAAAAAAHIAAAAAAQwAABtbnRyUkdCIFhZWiAAAAAAAAAAAAAAAABhY3NwAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAQAA9tYAAQAAAADTLQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlkZXNjAAAA8AAAACRyWFlaAAABFAAAABRnWFlaAAABKAAAABRiWFlaAAABPAAAABR3dHB0AAABUAAAABRyVFJDAAABZAAAAChnVFJDAAABZAAAAChiVFJDAAABZAAAAChjcHJ0AAABjAAAADxtbHVjAAAAAAAAAAEAAAAMZW5VUwAAAAgAAAAcAHMAUgBHAEJYWVogAAAAAAAAb6IAADj1AAADkFhZWiAAAAAAAABimQAAt4UAABjaWFlaIAAAAAAAACSgAAAPhAAAts9YWVogAAAAAAAA9tYAAQAAAADTLXBhcmEAAAAAAAQAAAACZmYAAPKnAAANWQAAE9AAAApbAAAAAAAAAABtbHVjAAAAAAAAAAEAAAAMZW5VUwAAACAAAAAcAEcAbwBvAGcAbABlACAASQBuAGMALgAgADIAMAAxADZWUDggBgEAAJALAJ0BKjoAOgA+bTCSRiQjIaEwFgkAgA2JQBWHOBs/zMoLISaWACRRJkfyBI4Lxdd/xd03yJpIGFTUqOQGP4xGt8WUYj9jng+v+2wjcftkQSQUlbocxVkWWLjln0a0IfovTZIAAP70062J9GjrH7g8QW5d3F+LQ8rXmHN3okrCe1/lyHE1N4J7O4QOlbIkk3Z32UNX6AwqF01B+DCejpwmvHfxAntOE/kxlbpt/hamu8MLs/k+nwpMp4g1BIvX+zQnii+y1Kl3Frd5sNA18WL4mwthjxSBIjPSiiMNEjmgFriUlUlYCmgWI0KnfUFVrqpG5rSDeBN77cUw84dmPu7Q3pMAAAA='/%3E%3C/svg%3E")",
            "backgroundPosition": "50% 50%",
            "backgroundRepeat": "no-repeat",
            "backgroundSize": "cover",
            "color": "transparent",
          }
        }
        width={176}
      />
    </div>
  </div>
</section>
`;
